### ISSUE TITLE
Throws System.FieldAccessException when using Injector in base class (Wrong IL generation for generics)

### ISSUE THREAD

**Issue by MAX-POLKOVNIK** [MAX-POLKOVNIK/DroidInjector#4]
```csharp
internal sealed class MySalesFragment : SalesFragment<MySalesFragment>
    {
        public static MySalesFragment NewInstance() => NewInstance("");
    }

    internal abstract class SalesFragment<T> : Fragment where T : SalesFragment<T>, new()
    {
        public static T NewInstance(string value) => new T();

        [View] private Button _myButton;

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            var view = inflater.Inflate(Resource.Layout.Main, container, false);

            Injector.InjectViews(view);

            return view;
        }
    }
```

```csharp
FragmentManager.BeginTransaction().Replace(Resource.Id.contentLayout, MySalesFragment.NewInstance())
                .Commit();
```

Will throw
```
UNHANDLED EXCEPTION:
System.FieldAccessException: Field `TestFieldAccessible.SalesFragment`1:_myButton' is inaccessible from method `TestFieldAccessible.SalesFragment`1<T_REF>:Polkovnik_DroidInjector_InjectViews (Android.Views.View)'

  at TestFieldAccessible.SalesFragment`1[T].OnCreateView (Android.Views.LayoutInflater inflater, Android.Views.ViewGroup container, Android.OS.Bundle savedInstanceState) [0x00010] in c:\users\polkovnik\documents\visual studio 2017\Projects\TestFieldAccessible\TestFieldAccessible\SalesFragment.cs:24 
  at Android.App.Fragment.n_OnCreateView_Landroid_view_LayoutInflater_Landroid_view_ViewGroup_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_inflater, System.IntPtr native_container, System.IntPtr native_savedInstanceState) [0x00020] in <5282a20cd8b04b8494a644a71f339169>:0 
  at (wrapper dynamic-method) System.Object.6e29bf92-8b8f-4c28-8900-dfedd51e5868(intptr,intptr,intptr,intptr,intptr)

```

**Comment by MAX-POLKOVNIK** [comment]
Cause wrong IL code generation for generics
```
View view = inflater.Inflate(2130903040, container, false);
((SalesFragment<>) this).Polkovnik_DroidInjector_InjectViews(view);
return view;
```

```
private void Polkovnik_DroidInjector_InjectViews(View view)
{
  Button viewById = (Button) view.FindViewById(2131034112);
  if (viewById == null)
    throw new InjectorException("Can't find view with ID 2131034112");
  ((SalesFragment<>) this)._myButton = viewById;
}
```

**Comment by MAX-POLKOVNIK** [comment]
Fixed in 0.1.4.1
